Announce CalVer switch (#2541)
conda-forge/conda-smithy: news/announce-calver.rst

**Added:**

* <news item>

**Changed:**

* IMPORTANT: This will be the last release of the ``3.x`` series.

>   In the next release, we will switch from Semver to CalVer, following a ``YY.MM.MICRO`` schema.

Author: I picked ``YY.MM.MICRO`` because that's what we follow in `conda`, but happy to pick another one. Some examples in https://calver.org

Reviewer: I'd much prefer to follow the style for the versioning of the pinning, since both appear prominently in rerender commits, and both are core cf infra. It makes sense to keep this consistent (though I'd be fine to replace the time-stamp portion with `MICRO`, or remove it entirely)

Besides, `YY` is not always obviously recognizable as CalVer, and doubly so if it's a three part version where one component is not actually date-related, e.g. 26.4.0 (BTW, in case that was implied, I'm definitely against leading zeros in `MM`; that's an abomination IMO).

I could kinda see an argument to follow conda's versioning scheme if people feel strongly about this, but again I think the coherence within conda-forge weighs stronger than coherence with conda.

Reviewer 2: If we use the pinnings, there is a practical issue about how the version is computed when we bump. That one is done to the second or microsecond. I'd rather stick with matching conda.

Reviewer: I addressed exactly that point. The pinning version looks like
```
2026.04.28.18.11.50
```
(guess I had conveniently forgotten that we have leading zeros there 🤦)

What I said was to remove the timestamp portion for smithy, either
```
2026.04.28  # if we're confident we won't release more than once a day
# or
2026.04.28.0
```

Reviewer 2: Ohhh I see. Yeah we should include a patch version. I am fine with that too.

Reviewer: I kinda wanna keep it simple and just use `2026.04.28` without the micro version.

If we ever need to release twice in a day for a bug fix (a rare but not unheard of occurrence), we could simply use the following day as a version - i.e. 2026.04.29 assuming a second release needs to be done right after 2026.04.28, even if it's still the 28th.

Author: So like this?

```suggestion
  In the next release, we will switch from Semver to CalVer, following a ``YYYY.MM.DD.MICRO`` scheme.
```

Author: Or this one, if we only add the patch bit when there's a patch release. IOW, we omit the explicit `.0`:

```suggestion
  In the next release, we will switch from Semver to CalVer, following a ``YYYY.MM.DD[.MICRO]`` scheme.
```

Author: OTOH, this may dictate how we implement the deprecation policy, so I'm inclined to not decide our schema in this news release (simply say "calver", scheme TBD):

```suggestion
  In the next release, we will switch from Semver to CalVer (concrete scheme to be decided).
```

Author: Upvote with emojis so we can see what we choose!

Reviewer: I'm fine with all three; out of your concrete suggestions, I'd prefer `YYYY.MM.DD[.MICRO]`, but my highest preference would be just `YYYY.MM.DD` (with same-day collisions resolved by using the next `DD`).

>   In the next release, we will switch from Semver to CalVer, following a ``YYYY.MM.DD[.MICRO]`` scheme.

Reviewer 2: ```suggestion
  In the next release, we will switch from Semver to CalVer, following a ``YYYY.MM.DD`` scheme.
```

  The first CalVer will be identical to the last SemVer release, except for the introduction

>   of a formal deprecation policy. (#2541)

Reviewer 2: ```suggestion
  of a formal 60-day deprecation policy. (#2541)
```
